core/Symbols.cc

```diff
@@ -1165,6 +1165,143 @@ TypePtr Symbol::sealedSubclassesToUnion(const GlobalState &gs) const {
     return result;
 }
 
+// Record a required ancestor for this class of module
+//
+// Each RequiredAncestor is stored into a magic method referenced by `prop` where:
+// * RequiredAncestor.symbol goes into the return type tuple
+// * RequiredAncestor.origin goes into the first argument type tuple
+// * RequiredAncestor.loc goes into the symbol loc
+// All fields for the same RequiredAncestor are stored at the same index.
+void Symbol::recordRequiredAncestorInternal(GlobalState &gs, Symbol::RequiredAncestor &ancestor, NameRef prop) {
```

Comment:
Could this method and `readRequiredAncestorsInternal` be functions defined in an anonymous namespace? Instead of relying on `this`, they could take a `Symbol *` argument called `self`.

Reply:
The problem is accessing `_locs` so I can `clear` it from the location we add by default.

Reply:
I was wondering after I left this comment if it was missing because I asked you to remove it, thanks! 😅